src/components/spacecraft/ShipInfo.ts

```diff
@@ -0,0 +1,163 @@
+import axios from 'axios';
+import getLetterList from './SpeciesArrays';
+
+export interface StapiResponse{
+  uid: string;
+  name: string;
+  numberOfDecks?: number | null;
+  crew?: string | null;
+  warpCapable: boolean;
+  mirror?: boolean;
+  alternateReality?: boolean;
+  activeFrom?: string | null;
+  activeTo?: string | null;
+  species: {
+    uid: string;
+    name: string;
+  } | null;
+}
+
+export const spaceCraftArray: StapiResponse[] = [];
+
+export default async function speciesSpacecraft() {
+  Object.values(speciesToShipMap).map(shipOfSpecies => shipOfSpecies.length=0);
+  for (let i = 0; i < 8; i+=1) {
+    await axios
+      .get(
+        "https://stapi.co/api/v2/rest/spacecraftClass/search?pageSize=100&pageNumber=" +
+          i
+      )
+      .then(response => {
+        filterShips(response.data.spacecraftClasses);
+        mapShips(spaceCraftArray);
+      });
+  }
+  return speciesToShipMap
+}
+
+function filterShips(response: StapiResponse[]){
+  spaceCraftArray.length=0;
+  for (let i=0; i<response.length; i+=1){
+    if(response[i].species!==null){
```

Comment:
Place a space between the if and opening parenthesis. Check other conditionals. Airbnb style guide 19.3